Compilation error with GCC 10.2.1 (embedded)

GCC version 10.2.1 
The `_jsonb_escape` function returns a log, but this is assigned to a `jsonbcode` eum value in the calling routines.
It isn't always possible to add a compilation flag to allow this, as the compiler error suggests:


```json-build/json-build.h: In function 'jsonbcode jsonb_key(jsonb*, char*, size_t, const char*, size_t)':

json-build/json-build.h:406:28: error: invalid conversion from 'long int' to 'jsonbcode' [-fpermissive]

  406 |         ret = _jsonb_escape(&pos, buf + b->pos, bufsize - b->pos, key, len);
      |               ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                            |
      |                            long int
```

--- a/json-build.h
+++ b/json-build.h
@@ -327,7 +327,7 @@ jsonb_object_pop(jsonb *b, char buf[], size_t bufsize)
     return code;
 }
 
-static long
+static jsonbcode
 _jsonb_escape(
     size_t *pos, char buf[], size_t bufsize, const char str[], size_t len)
 {

--- a/test/Makefile
+++ b/test/Makefile
@@ -1,5 +1,5 @@
 TOP = ..
-CC ?= gcc
+CC = cc
 
 EXES = test fuzz
 

